consul/src/main/scala/io/buoyant/consul/CatalogNamer.scala

```diff
@@ -2,16 +2,30 @@ package io.buoyant.consul
 
 import com.twitter.finagle._
 import com.twitter.util._
-import io.buoyant.consul.v1.{UnexpectedResponse, ServiceNode}
+import io.buoyant.consul.v1.{ServiceNode, UnexpectedResponse}
+import io.buoyant.namer.Metadata
 
 class CatalogNamer(
   idPrefix: Path,
-  api: v1.CatalogApi,
-  includeTag: Boolean = false
+  catalogApi: v1.CatalogApi,
+  agentApi: v1.AgentApi,
+  includeTag: Boolean = false,
+  setHost: Boolean = false
 ) extends Namer {
 
   import CatalogNamer._
 
+  private[this] lazy val domainFuture: Future[String] =
```

Comment:
Something to note: the domain request will be fired once when the catalog namer does `init`, and then never again.  This means if that request fails for any reason, the domain will never be populated.  Worse, anything flatMapped from the domain future (like the rest of the body of `init` will never be executed).  

Similarly, this also means that we will never pick up changes if the domain value in consul changes.

Does the consul self API support long polling?  Can we have the domain be populated by a Var.async instead?

@olix0r any thoughts on this?